src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.Windows.Export.cs

```diff
@@ -102,6 +90,165 @@ public void MoveTo(X509Certificate2Collection collection)
             }
         }
 
+        public byte[] ExportPkcs12(Pkcs12ExportPbeParameters exportParameters, SafePasswordHandle password)
+        {
+            bool tryAes256sha256 = exportParameters is
+                Pkcs12ExportPbeParameters.Default or
+                Pkcs12ExportPbeParameters.Pbes2Aes256Sha256;
+
+            byte[] exported = ExportPkcs12(tryAes256sha256, password, out bool aes256Sha256);
+
+            // What we asked for and what we got are the same - so return it as-is.
+            if (tryAes256sha256 == aes256Sha256)
+            {
+                return exported;
+            }
+
+            if (!tryAes256sha256)
+            {
+                Debug.Fail("3DES PKCS12 export failed.");
+                throw new CryptographicException();
+            }
+
+            return ReEncryptAndSealPkcs12(exported, password, Helpers.WindowsAesPbe);
+        }
+
+        public byte[] ExportPkcs12(PbeParameters exportParameters, SafePasswordHandle password)
+        {
+            byte[] exported = ExportPkcs12(preferAes256Sha256: true, password, out _);
+            return ReEncryptAndSealPkcs12(exported, password, exportParameters);
+        }
+
+        private unsafe byte[] ExportPkcs12(bool preferAes256Sha256, SafePasswordHandle password, out bool aes256Sha256)
+        {
+            Interop.Crypt32.DATA_BLOB dataBlob = new Interop.Crypt32.DATA_BLOB(IntPtr.Zero, 0);
+            Interop.Crypt32.PFXExportFlags flags =
+                Interop.Crypt32.PFXExportFlags.EXPORT_PRIVATE_KEYS |
+                Interop.Crypt32.PFXExportFlags.REPORT_NOT_ABLE_TO_EXPORT_PRIVATE_KEY;
+
+            Interop.Crypt32.PKCS12_PBES2_EXPORT_PARAMS* exportParams = null;
+            aes256Sha256 = preferAes256Sha256 && s_supportsAes256Sha256;
+
+            if (aes256Sha256)
+            {
```

Comment:
Should the && s_supports check be here instead of in the caller?  Or maybe assert s_supports in the if?

Reply:
It is there on line 130. `aes256Sha256` is `preferAes256Sha256 && s_supportsAes256Sha256`. If this was not clear because of the variable names, I am open to suggestions to improve it (Or perhaps I am still misunderstanding your point).

Reply:
I must have just scrolled up and down too many times... I thought the s_supports check was in the calling method, not here.  My bad.